### PR TITLE
Add Ctest support for functional testing

### PR DESCRIPTION
**Pull request by oskirby**
## Description
This is mostly just a minor annoyance that every time I have to run a functional test I have to remember and run a bunch of commands manually to get the functional testing envrionment setup. This makes it so that we can run a tests with something like the following:

 - `ctest -L functional`: Run all the functional tests
 - `ctest -L unit`: Run all the unit tests.
 - `ctest -L auth`: Run the authentication tests.
 - `ctest -L qml`: Run the QML tests.
 - `ctest -R <test name>`: Run a single test.

## Reference

    i.e Jira or Github issue URL

## Checklist
    
- [ ] My code follows the style guidelines for this project
- [ ] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [ ] I have performed a self review of my own code
- [ ] I have commented my code PARTICULARLY in hard to understand areas
- [ ] I have added thorough tests where needed

### CODE REVIEW

**Review by strseb — APPROVED** [review]
Can we add the tag to `mz_add_test_target` aswell ? :)

**Comment by oskirby** [comment]
Uh oh, looks like the socksproxy tests don't pass on MacOS. Guess I should fix that first.

**Review by mcleinman — APPROVED** [review]
👍🏻 Thanks!

This seems reasonable, though this area is not my expertise. I'd encourage waiting for at least one other approval.

**Pull request merged**
